retrieval/scrape.go

```diff
@@ -446,6 +461,28 @@ func (sl *scrapeLoop) stop() {
 	<-sl.done
 }
 
+func (sl *scrapeLoop) processScrapeResult(samples model.Samples, scrapeErr error, start time.Time) error {
+	// Collect samples post-relabelling and label handling in a buffer.
+	buf := &bufferAppender{buffer: make(model.Samples, 0, len(samples))}
```

Comment:
I'm actually in the middle of investigating heap issues. This will not be a problem.

We need this anyway to calculate the number of post-metric relabelling samples.